dotCMS/src/main/java/com/dotcms/rest/api/v1/folder/FolderResource.java

```diff
@@ -139,6 +145,97 @@ public final Response loadFolderAndSubFoldersByPath(@Context final HttpServletRe
+     * 		subfolder1
+     * 	testfolder3
+     *
+     * default_copy
+     * 	folder1_copy
+     * 		subfolder1_copy
+     * 		subfolder2_copy
+     *                  testsubfolder3_copy
+     * 	folder2_copy
+     * 		subfolder1_copy
+     * 	testfolder3_copy
+     *
+     * 	Value Sent | Expected Result
+     * ------------ | -------------
+     * //default/ | folder1, folder2, testfolder3
+     * //default/fol | folder1, folder2
+     * //default/fol/ | Nothing
+     * //default/bla | Nothing
+     * //default/folder1/ | subfolder1, subfolder2, testsubfolder3
+     * //default/folder1/s | subfolder1, subfolder2
+     * //default/folder1/b | Nothing
+     * / | folder1,folder2,testfolder3,folder1_copy,folder2_copy,testfolder3_copy
+     * /f | folder1,folder2,folder1_copy,folder2_copy
+     * /folder1/ | subfolder1, subfolder2, testsubfolder3,folder1_copy,folder2_copy,testfolder3_copy
+     * /folder1/s | subfolder1, subfolder2,subfolder1_copy, subfolder2_copy
+     * f | folder1,folder2,folder1_copy,folder2_copy
+     * folder1/ | subfolder1, subfolder2, testsubfolder3,folder1_copy,folder2_copy,testfolder3_copy
+     * folder1/s | subfolder1, subfolder2,subfolder1_copy, subfolder2_copy
+     *
+     *
+     * @param searchByPathForm path to look for the sub-folders
+     * @return List of {@link FolderSearchResultView}
+     * @throws DotDataException
+     * @throws DotSecurityException
+     */
+    @POST
+    @Path ("/byPath")
+    @JSONP
+    @NoCache
+    @Produces({MediaType.APPLICATION_JSON})
+    public final Response findSubFoldersByPath(@Context final HttpServletRequest httpServletRequest,
+            @Context final HttpServletResponse httpServletResponse,
+            final SearchByPathForm searchByPathForm
+            ) throws  DotDataException, DotSecurityException   {
+
+        final InitDataObject initData =
+                new WebResource.InitBuilder(webResource)
+                        .rejectWhenNoUser(true)
+                        .requiredBackendUser(true)
+                        .requiredFrontendUser(false)
+                        .requestAndResponse(httpServletRequest, httpServletResponse)
+                        .init();
 
+        final User user = initData.getUser();
+
+        String path = searchByPathForm.getPath();
+        String siteId = null;
+        String folderPath = path;
+
+        if(path.startsWith(StringPool.DOUBLE_SLASH)){
+            //Removes // to search the host
+            path = path.startsWith(StringPool.DOUBLE_SLASH) ? path.substring(2) : path;
+            final String sitePath = path.split(StringPool.FORWARD_SLASH,2)[0];
+            final Host site = APILocator.getHostAPI().findByName(sitePath, user,false);
+            if(null == site){
+                throw new DoesNotExistException(String.format(" Couldn't find any host with name `%s` ",sitePath));
+            }else{
+                siteId = APILocator.getHostAPI().findByName(sitePath, user,false).getIdentifier();
```

Comment:
why ask again for the site, would be not just site.getIdentifier

Reply:
done